### PR TITLE
feat: add get_collections api

### PR DESCRIPTION
**Pull request by navinkarkera**
Adds `get_collections` api for https://github.com/openedx/edx-platform/pull/35324

### CODE REVIEW

**Review by rpenido — CHANGES_REQUESTED** [review]
@navinkarkera Could you add a test for this new function?

**Review by rpenido — APPROVED** [review]
LGTM :+1:
Thank you for your work, @navinkarkera !

- [x] I tested this using the instructions from https://github.com/openedx/edx-platform/pull/35324
- [x] I read through the code
- [x] ~I checked for accessibility issues~
- [x] Includes documentation

@pomegranited This is ready for CC review!

**Review by pomegranited — APPROVED** [review]
@navinkarkera :+1: with one nit/question for @ormsbee , but I'm fine with merging this as-is.

- [x] I tested this using the instructions on https://github.com/openedx/edx-platform/pull/35324
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] Includes documentation
- [x] <strike>User-facing strings are extracted for translation</strike> N/A

File: openedx_learning/apps/authoring/collections/api.py [hunk 1]
```diff
@@ -78,3 +79,13 @@ def get_learning_package_collections(learning_package_id: int) -> QuerySet[Colle
     return Collection.objects \
                      .filter(learning_package_id=learning_package_id, enabled=True) \
                      .select_related("learning_package")
+
+
+def get_collections(enabled: bool | None = None) -> QuerySet[Collection]:
+    """
+    Get all collections, optionally caller can filter by enabled flag
+    """
+    qs = Collection.objects.all()
```
[the remark this answers is not in the document]
Comment (navinkarkera) [reply openedx_learning/apps/authoring/collections/api.py:88]: @pomegranited I agree about making tests simpler, as you can see I had to make use of `self.assertQuerySetEqual` as the order was not matching in CI.

[the remark this answers is not in the document]
Comment (navinkarkera) [reply openedx_learning/apps/authoring/collections/api.py:89]: Updated PR to order by `pk`, we can remove the commit if @ormsbee does not agree.

[the remark this answers is not in the document]
Comment (ormsbee) [reply openedx_learning/apps/authoring/collections/api.py:89]: I originally made it sort by `pk` as a cheap, lazy way of sorting it by creation time. But I think it's a reasonable default sort in a lot of situations, so this is fine by me.

**Comment by openedx-webhooks** [comment]
@navinkarkera 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-learning&entry.867055334=https://github.com/openedx/openedx-learning/pull/214&entry.1484655318=https://github.com/navinkarkera&entry.752974735=2024-08-19+14:49&entry.1917517419=2024-08-27+04:31&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**